### PR TITLE
Write error message if upload fails and switch to redcap_write_oneshot

### PR DESCRIPTION
**Pull request by ljwoodley**
Switched to `redcap_write_oneshot` as `redcap_write` does not return a list with success or outcome_message when the upload fails. Which is weird as the documentation says a list should be returned.

I did not update `load_screen_into_screen_follow_up.R` as I'm unsure if `redcap_write_oneshot` will negatively affect upload.

### CODE REVIEW

File: load_person_data_into_results.R [hunk 1]
```diff
@@ -24,7 +24,7 @@ Sys.getenv("INSTANCE")
 uf_person_data <- get_person_data(dev_mode) %>%
         mutate(uf_identifiers_and_determination_complete = '2')
 
-write_data <- redcap_write(
+write_data <- redcap_write_oneshot(
```
Comment (pbchase) [inline load_person_data_into_results.R:27]: That's a 122k record write on a new system. I can't recommend `redcap_write_oneshot` here

**Comment by pbchase** [comment]
> I did not update `load_screen_into_screen_follow_up.R` as I'm unsure if `redcap_write_oneshot` will negatively affect upload.

that's a 22K record write event. I would not use `redcap_write_oneshot` in `load_screen_into_screen_follow_up.R`

File: load_person_data_into_results.R (hunk 1 above)
Comment (ChemiKyle) [reply load_person_data_into_results.R:27]: `write_data` definitely contains a `success` object here

**Comment by ChemiKyle** [comment]
`redcap_write` contains all of the objects contained in `redcap_write_oneshot` with the exception of `raw_text`.  
I imagine there was an error with at least one record in `redcap_write`, adding `continue_on_error = T` as an arg to `redcap_write` will probably "solve" the issue.

File: load_person_data_into_results.R (hunk 1 above)
Comment (ljwoodley) [reply load_person_data_into_results.R:27]: yes but if it fails there's no list returned. I guess we can just say "upload failed" in the email body if it fails. Only downside is that we would not get the exact reason for failure.

Comment (pbchase) [reply load_person_data_into_results.R:27]: True, but we have still improved the process.

Comment (ljwoodley) [reply load_person_data_into_results.R:27]: ok, I have a function that does that. I'll make the changes.

**Comment by ljwoodley** [comment]
> [`redcap_write`](https://rdrr.io/cran/REDCapR/man/redcap_write.html)'s response contains all of the elements contained in the response for [`redcap_write_oneshot`](https://rdrr.io/cran/REDCapR/man/redcap_write_oneshot.html) with the exception of `raw_text`.
> I imagine there was an error with at least one record in `redcap_write`, adding `continue_on_error = T` as an arg to `redcap_write` will probably "solve" the issue.

Yes, this returns a list but success is TRUE, even if the upload is not successful.

**Comment by ChemiKyle** [comment]
Interesting. I suppose the proper solution is to extract the record(s) causing failure and continue with `redcap_write`, but that feels a lot like reinventing the wheel.

**Comment by ChemiKyle** [comment]
Looks good, merging.

**Pull request merged**